File: EmpoweredPixels/ClientApp/src/app/match/+models/paging-options.ts
export class PagingOptions
{
    public pageNumber = 1;
    public pageSize = 10;
    public stringFilters: StringFilter[] = new Array();
    public numberFilters: NumberFilter[] = new Array();
    public sortSetting?: SortSetting;
}

export class StringFilter
{
    query?: string;
    propertyName?: string;
}

export class NumberFilter
{
    propertyName?: string;
    lowerbound?: number;
    upperbound?: number;
}

export class SortSetting
{
    propertyName?: string;
    descending?: boolean;
}

File: EmpoweredPixels/Utilities/Paging/Models/PagingOptions.cs
using System.Collections.Generic;

namespace EmpoweredPixels.Utilities.Paging
{
  public class PagingOptions
  {
    public int PageNumber { get; set; }

    public int PageSize { get; set; }

    public ICollection<StringFilter> StringFilters { get; set; } = new List<StringFilter>();

    public ICollection<NumberFilter> NumberFilters { get; set; } = new List<NumberFilter>();

    public SortSetting SortSetting { get; set; }
  }
}
